wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsTopologyClient.java

```diff
@@ -227,6 +232,100 @@ private void initWsTopologyStub(Connection connection)
             log.error(initErrMessage);
     }
 
+    static public enum TopologyGroupQueryKind
+    {
+        PLANE,
+        HTHOR,
+        THOR,
+        ROXIE;
+
+        public String getText() throws Exception
+        {
+            switch (this)
+            {
+            case PLANE:
+                return "Plane";
+            case HTHOR:
+                return "hthor";
+            case THOR:
+                return "Thor";
+            case ROXIE:
+                return "Roxie";
+            default:
+                throw new Exception("Unknown TopologyGroupQueryKind");
+            }
+        }
+    }
+
+    /**
+     * Fetch data storage planes - assumes Containerized Plane filter
+     *
+     * @return wrapped response object
+     * @throws HpccContainerizedUnsupportedException if targeting baremetal HPCC environment
+     * @throws Exception if error encountered
+     */
+    public TpGroupQueryResponseWrapper getDataStoragePlanes() throws HpccContainerizedUnsupportedException, Exception
+    {
+        if (!isTargetHPCCContainerized())
+            throw new HpccContainerizedUnsupportedException("getDataStoragePlanes(): Only supported in CONTAINERIZED mode");
+
+        return getDataStoragePlanes(TopologyGroupQueryKind.PLANE.getText());
+    }
+
+    /**
+     * Fetch data storage planes
+     *
+     * @param kind enumerated options
+     * @return wrapped response object
+     * @throws HpccContainerizedUnsupportedException if kind not provided while targeting baremetal HPCC environment
+     * @throws Exception if error encountered
+     */
+    public TpGroupQueryResponseWrapper getDataStoragePlanes(TopologyGroupQueryKind kind) throws HpccContainerizedUnsupportedException, Exception
```

Comment:
Should this getDataStoragePlanes() (and the one below) be renamed to getTopologyGroups()? Does ‘Data Storage Plane’ make sense while targeting a baremetal environment?

Reply:
Good point, now that it is no longer Containerized only function, the name should reflect the new functionality...